csrc/ops/indexing.cpp

```diff
@@ -100,6 +100,32 @@ TensorView* indexSelect(
   return out;
 }
 
+// This is a restricted version of torch.index_put(..., accumulate=true)
+TensorView* indexAccumulate(
+    TensorView* acc_tv,
+    TensorView* index_tv,
+    TensorView* value_tv) {
+  DataType dtype = acc_tv->getDataType().value();
+  NVF_CHECK(
+      dtype != DataType::Null, "Invalid datatype provided for new value.");
+
+  std::vector<IterDomain*> acc_domain =
+      TensorDomain::noReductions(acc_tv->getLogicalDomain());
+  std::vector<IterDomain*> index_domain =
+      TensorDomain::noReductions(index_tv->getLogicalDomain());
+  std::vector<IterDomain*> value_domain =
+      TensorDomain::noReductions(value_tv->getLogicalDomain());
+
+  // TODO: add shape checks and handle corner cases later
+  NVF_CHECK(acc_domain.size() == 2);
+  NVF_CHECK(index_domain.size() == 1);
+  NVF_CHECK(value_domain.size() == 2);
```

Comment:
From your comments above:
```
    // Producers:
    //     accumulate [ vocab, hidden ]
    //     index [ * ]
    //     value [ *, hidden ]
    // Consumers:
    //     output [ vocab, hidden ]
```
This conveys that index can be n-D (> 1) and value is > 2-D?

Reply:
Yeah, I think the op in its general form should be able to support that.

I put the assert here explicitly just to keep thing simple. But there's no good reason to do that. Let me revert this.

Reply:
@jjsjann123 do you plan on adding support for the general case here? Are there any complications in supporting the general case?

Reply:
Not in this PR (or even the follow up, which will be a codegen PR).

I want to get codegen out first for the simple case and then worry about expanding the support.

I'm not totally sure about complexities here, mostly regarding indexing on `index_tv`. That part should be similar to how `indexSelect` is handling it and it has a requirement on `index_tv` being 1D.

So I think I will keep the restrictions here until I figure out those obstacles first. Besides, for integration, we can always rely on reshape inserted by the integration to have a functional support, even though the codegen support is limited.